### DISCUSSION TITLE
Ploting two fibers of grains orientations in the same map.

### DISCUSSION THREAD

**Discussion by M-Eddahbi** [mtex-toolbox/mtex#1259] [Ask Anything] [answered]
Dear Rüdiger

I tried to reveal in the same map two population of grains orientations; those lying in fiber 1 (f1) and those in fiber 2 (f2). I employed the following syntaxes: 

% fiber 1

f1 = fibre(Miller(0,0,1,CS1), zvector);
colorKey1 = ipfSpotKey(CS1);
colorKey1.inversePoleFigureDirection = f1.r;
colorKey1.center = f1.h;
colorKey1.color = [1 0 0]; 
colorKey1.psi = S2DeLaValleePoussin('halfwidth',7.5*degree);

% fiber 2

f2 = fibre(Miller(1,1,0,CS1), zvector);
colorKey2 = ipfSpotKey(CS1);
colorKey2.inversePoleFigureDirection = f2.r;
colorKey2.center = f2.h;
colorKey2.color = [0 1 0]; 
colorKey2.psi = S2DeLaValleePoussin('halfwidth',7.5*degree);

figure, plot(grains, colorKey1.orientation2color(grains.meanOrientation))
hold on
plot(grains ,colorKey2.orientation2color(grains.meanOrientation))

However, when I execute the above syntaxis, only the grains of f1 appears in the output plot-map. I repreated the exercise but in vain.

1.	Is there some think wrong in the above syntaxes?
2.	Maybe, it is preferable to define the grains of each fiber and then plot them. But I didn’t found the way to define the grains of each fiber ?. How can I do that?

Thank you very much in advance for your help

M. Eddahbi

**Comment by kilir** [comment, marked as answer]
Hi,
I would have expected only those of f2 to show up - which is the expected behaviour since you create two colormaps where everything outside your interest is white. Hence you plot white grains on top of the ones you colored with key 1.

Here's an example hot to combine both:
```
mtexdata ferrite
grains =ebsd.calcGrains


CS = ebsd('F').CS

ck = ipfSpotKey(CS);
ck.inversePoleFigureDirection = zvector;
ck.center = [Miller(0,0,1,CS1) Miller(1,1,0,CS1)];
ck.color = [1 0 0; 0 1 0];
ck.psi = S2DeLaValleePoussin('halfwidth',7.5*degree);

plot(grains('f'), ck.orientation2color(grains('f').meanOrientation))
nextAxis
plot(ck)
```

Alternatively, you can select grains which are within a certain threshold to a fibre `grains(angle(f1,grains.meanOrientation) < 10*degree)`

Hope that helps.
Cheers,
Rüdiger

**Reply by M-Eddahbi** [reply]
Thank you very much Rüdiger. Your suggestion works.
Thanks is also addressed to Kilir.
